mbed_devices/_internal/candidate.py

```diff
@@ -1,13 +1,31 @@
 """Defines Candidate model used for device detection."""
-from typing import Optional, NamedTuple
+from collections import namedtuple
+from typing import Optional, List
 from pathlib import Path
 
 
-class Candidate(NamedTuple):
-    """Device connected to the host computer."""
+class Candidate(namedtuple("Candidate", ["product_id", "vendor_id", "mount_points", "serial_number", "serial_port"])):
```

Comment:
It might've been easier to use `dataclass`es here rather than trying to work with `NamedTuple`s weird behaviour. However, `dataclass` is only part of the stdlib in 3.7+. I'm wondering if we should just make this a regular `class`? I'm not sure we need the `NamedTuple` inheritance? I know it buys us immutability but I'm not sure if it's worth it to inherit the rest of `NamedTuple`s hacky weirdness in this case.